app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/RadioGroup.kt

```diff
@@ -0,0 +1,905 @@
+      style = style,
+      color = textColor,
+      modifier = Modifier,
+    )
+  }
+}
+
+@Composable
+private fun VerticalRadioGroup(
+  data: List<OptionData>,
+  radioGroupStyle: Vertical,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Column(modifier) {
+    for (radioOptionData in data) {
+      val radioOptionStyle = when (radioGroupStyle) {
+        is Vertical.Default -> Default
+        is Vertical.Icon -> RadioOptionStyle.Icon(radioOptionData.iconResource ?: Painter(R.drawable.pillow_hedvig)) // todo: how is that for placeholder
+        is Vertical.Label -> Label(radioOptionData.labelText ?: "") // todo: no placeholder
+        is Vertical.LeftAligned -> LeftAligned
+      }
+      RadioOption(
+        data = radioOptionData,
+        radioOptionStyle = radioOptionStyle,
+        groupLockedState = groupLockedState,
+        radioOptionSize = radioGroupSize.toOptionSize(),
+        onOptionClick = {
+          onOptionClick(radioOptionData.id)
+        },
+      )
+      Spacer(Modifier.height(4.dp))
+    }
+  }
+}
+
+@Composable
+private fun VerticalRadioGroupWithLabel(
+  data: List<OptionData>,
+  radioGroupStyle: VerticalWithGroupLabel,
+  groupLockedState: LockedState,
+  radioGroupSize: RadioGroupSize,
+  contentPaddingValues: PaddingValues,
+  onOptionClick: (String) -> Unit,
+  modifier: Modifier = Modifier,
+) {
+  Surface(
+    shape = radioGroupSize.getShape(),
+    color = radioOptionColors.containerColor,
+  ) {
+    Column(
+      modifier,
+    ) {
+      val labelTextColor = radioOptionColors.labelTextColor(groupLockedState)
+      HedvigText(
+        modifier = Modifier.padding(contentPaddingValues),
+        text = radioGroupStyle.groupLabelText,
+        style = MediumSizeRadioOptionTokens.LabelTextFont.value, // same for all sizes in figma
+        color = labelTextColor,
+      )
+      Column(modifier) {
+        for (radioOptionData in data) {
+          val interactionSource = remember { MutableInteractionSource() }
+          val modifierRipple = Modifier
+            .clickable(
+              enabled = calculateLockedStateForItemInGroup(radioOptionData, groupLockedState) == NotLocked,
+              role = Role.RadioButton,
+              interactionSource = interactionSource,
+              indication = ripple(
+                bounded = true,
+              ),
+              onClick = {
+                onOptionClick(radioOptionData.id)
+              },
+            )
+          val radioOptionStyle = when (radioGroupStyle) {
+            is VerticalWithGroupLabel.Default -> Default
+            is VerticalWithGroupLabel.Icon -> RadioOptionStyle.Icon(
+              radioOptionData.iconResource ?: Painter(R.drawable.pillow_hedvig),
+            ) // todo: how is that for placeholder
+            is VerticalWithGroupLabel.Label -> Label(radioOptionData.labelText ?: "") // todo: no placeholder
```

Comment:
In which scenarios would it be possible to be in the label style but have no labelText? Would it make sense for that to be impossible in the first place?

Reply:
I'll try one thing!